src/components/authentication/SigninGithubWrapper.vue

```diff
@@ -0,0 +1,529 @@
+
+        console.log(url);
+        await window.ipc.invoke(IPC_HANDLERS.FILE_SYSTEM, {
+          func: IPC_FUNCTIONS.OPEN_EXTERNAL_LINK,
+          data: url,
+        });
+
+        const wait = function (period) {
+          return new Promise((resolve) => {
+            setTimeout(resolve, period);
+          });
+        };
+
+        const pollForToken = async (creds) => {
+          await wait(3500);
+          const tokenURL = `${yattURL}/v1/app/oauth/github/token/${tokenId}`;
+          let header = {};
+          if (creds?.yatt?.length > 0 && creds?.yatt[0]?.accessToken) {
+            header.headers = {
+              Authorization: `Bearer ${creds.yatt[0].accessToken}`,
+              Accept: "application/json",
+            };
+          }
+
+          for (let count = 0; count < 30; count++) {
+            let response;
+            try {
+              response = await axios.get(tokenURL, header);
+            } catch (err) {
+              response = err.response;
+            }
+
+            if (response.status === 200) {
+              return response;
+            }
+            await wait(1500);
+          }
+          return { status: 500, message: "Timed out polling for token." };
+        };
+        let finalResponse = await pollForToken(this.credentials);
+        if (finalResponse.status !== 200) {
+          this.loading = false;
+          this.$root.$emit("overlay", false);
+          this.snackBar.enabled = true;
+          this.snackBar.message = finalResponse.message
+            ? finalResponse.message
+            : this.$tc("caption.log_in_failed", 1);
+        } else {
+          if (finalResponse?.data?.github) {
+            finalResponse.data.github.yattOauthTokenId = tokenId;
+            finalResponse.data.github.type = "oauth";
+          }
+          if (finalResponse?.data?.yatt) {
+            finalResponse.data.yatt.oauthTokenIds = [tokenId];
+          } else {
+            // If we don't get any YATT data back, then we must have used an
+            //   API key for access, so just update our current creds with the
+            //   new oauth token ID and pass them along to be saved.
+            let tempCredentials = this.credentials;
+            if (tempCredentials.yatt[0].oauthTokenIds) {
+              tempCredentials.yatt[0].oauthTokenIds.push(tokenId);
+            } else {
+              tempCredentials.yatt[0].oauthTokenIds = [tokenId];
+            }
+            await window.ipc.invoke(IPC_HANDLERS.DATABASE, {
+              func: IPC_FUNCTIONS.UPDATE_CREDENTIALS,
+              data: tempCredentials,
+            });
+          }
+          this.postLogin(finalResponse.data);
+        }
+      }
+    },
+    async signInWithApiKey() {
+      this.postLogin({
+        jira: {
+          type: "basic",
+          accessToken: Buffer.from(this.userName + ":" + this.apiKey).toString(
+            "base64"
+          ),
+          url: this.instanceUrl,
+        },
+      });
+    },
```

Comment:
Did you test this?  It looks to me like GH uses a bearer token, not basic auth.

https://docs.github.com/en/github-ae@latest/rest/authentication/authenticating-to-the-rest-api#basic-authentication

Reply:
Sorry API key isn't supported here yet. I've removed that method along with the others that aren't currently supported.